realm/realm-library/src/main/java/io/realm/RealmInteger.java

```diff
@@ -50,111 +51,70 @@ public RealmInteger(String value) {
      *
      * @param newValue new value.
      */
-    public void set(long newValue) {
-        value = newValue;
-    }
+    public abstract void set(long newValue);
 
     /**
      * Increments the {@code RealmInteger}, adding the value of the argument.
      * Increment/decrement from all devices are reflected in the new value, which is guaranteed to converge.
      *
      * @param inc quantity to be added to the counter.
      */
-    public void increment(long inc) {
-        value += inc;
-    }
+    public abstract void increment(long inc);
 
     /**
      * Decrements the {@code RealmInteger}, subtracting the value of the argument.
      * Increment/decrement from all devices are reflected in the new value, which is guaranteed to converge.
      *
      * @param dec quantity to be subtracted from the counter.
      */
-    public void decrement(long dec) {
-        value -= dec;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public byte byteValue() {
-        return (byte) value;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public double doubleValue() {
-        return (double) value;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public float floatValue() {
-        return (float) value;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public int intValue() {
-        return (int) value;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public long longValue() {
-        return value;
-    }
+    public abstract void decrement(long dec);
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public short shortValue() {
-        return (short) value;
+    public String toString() {
+        return String.valueOf(longValue());
     }
 
     /**
-     * {@inheritDoc}
+     * RealmIntegers compare strictly by their values.
+     * Final to ensure contract over all subclasses
```

Comment:
It was more the `final to ensure contract` comment. It sounds like an internal comment since marking it final kinda already means users cannot override it?